contracts/DAOTreasury.cdc

```diff
@@ -50,10 +53,57 @@ pub contract DAOTreasury {
       wants. This means it's very imporant for the signers
       to know what they are signing.
     */
-    pub fun executeAction(actionUUID: UInt64) {
+    pub fun executeAction(actionUUID: UInt64, signaturePayload: MyMultiSig.MessageSignaturePayload) {
+      self.validateTreasurySigner(identifier: actionUUID.toString(), signaturePayload: signaturePayload)
+
       let selfRef: &Treasury = &self as &Treasury
       self.multiSignManager.executeAction(actionUUID: actionUUID, {"treasury": selfRef})
-      emit ExecuteAction(actionUUID: actionUUID)
+      emit ExecuteAction(actionUUID: actionUUID, proposer: signaturePayload.signingAddr)
```

Comment:
could you help me understand why there is a need to add proposer to Action, if proposer is already available in `signaturePayload.signingAddr`?
https://github.com/DapperCollectives/VESSEL/pull/50/files#diff-d6c67e336e7c8eb3ba7d49cd0a16be12910ac98006b41b6abfddf320a36295f1R31

Reply:
Might be useful in case we want in the future to display who proposed the action. For example script get_proposed_actions.cdc returns the intents, this might be expanded to support the proposers too. ExecuteAction here is an event (naming can be confusing I know :D) that is emitted only once when the action is successfully executed
Looping in @jacksonConrad for thoughts

Reply:
I see. As the ticket is related to restrict action to signers, I was wondering why we need to add `proposer` to Action. So looks like its two different items - Adding proposer to action is purely extending info in Action. :)